File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int n = arr.size();
        int lo = 1;
        int hi = n-2;
        int idx = -1;
        while(lo<=hi){
            int mid = lo +(hi-lo)/2;
            if(arr[mid]>arr[mid-1]){
                if(arr[mid]>arr[mid+1]){
                     idx = mid;
                     break;
                     }
                else lo = mid+1;
            }
          else if(arr[mid]>arr[mid+1]) hi = mid-1;
        }
    return idx;
    }           
};

File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int n = arr.length;
        int lo = 0;
        int hi = n-1;
        while(lo<=hi){
            int mid = lo + (hi-lo)/2;
            if(mid==0) return hi;
            else if(mid==n-1) return lo;
            else if(arr[mid]>arr[mid-1] && arr[mid]>arr[mid+1]) return mid;
            else if(arr[mid]>arr[mid-1] && arr[mid]<arr[mid+1]) lo = mid+1;
            else hi = mid-1;
        }
        return 1;
    }
}